### PR TITLE
Removes vampire blood costs, increases vampire cooldowns of abilities that used to cost blood.

### PR DESCRIPTION
**Pull request by Arzion5**
* experiment: Vampires have been slightly reworked!
 * tweak: Vampire abilities no longer cost blood.
 * tweak: All vampire abilities that used to have a cost had their cooldown increased.
 * tweak: Very slight description changes 

:cl:
This is more of an experimental PR. It's not properly done and it's a bit shoddy, but it should work. Personally, I want the community's opinion and coding advice. Vampires are controversial antagonists because they tend to do nothing for most of the round, players having dubbed it "vampstended". I believe that this can be attributed to a lack of confidence for vampires. It may be attributed to the following:
- Lack of powers as well as the give-away that you are a vampire when using the abilities.
- Cannot drink blood from dead people, which is an issue that was likely unintended, but I do not know how to fix.
- Some roles may have a very limited access to maintenance, or being in places that are too frequented by other players.
- Rounds with an AI and security personnel, you cannot kill or strip security personnel without setting off their secure GPS, while silicons are allowed to kill you on the default Asimov lawset, as well as being immune to most of your abilities.
- The only area of effect attacks available are both stuns, so you are at a massive disadvantage if a group comes after you. This is not a really viable design for solitary antagonists.

The purpose of the change is to encourage vampires to act without being restricted by the active blood mechanic, because they cannot drink from corpses. Personally, I would have wanted to turn vampires into a potential mid-round antagonist, they are unfit to have a fully dedicated antagonist role. However, I advise increasing the amount of players required until vampire is a gamemode that can be rolled, but adding this change to the PR would turn it into a bundle. It will become a separate PR at some point.
As I have mentioned, the changes are experimental, serving as both a way to assess the public opinion on vampires and agreed ways to make them better as well as hopefully receive feedback related to coding.
As for the list related to the vampire abilities, the cooldowns are as following:
- Hypnotise from 3 minutes to 4 minutes.
- Diseased touch from 3 minutes to 5 minutes.
- Chiroptean Screech from 3 minutes to 4 minutes.
- Enthrall from 3 minutes to 10 minutes.
- Summon Bats from 2 minutes to 4 minutes.
- Bat form from 1 minute to 1 minute and 30 seconds.
- Shadowstep from 2 seconds to 10 seconds. (I believe that the previous cooldown was 20 seconds, I will make the proper adjustments if this is the case.)

### CODE REVIEW

**Comment by ShiftyRail** [comment]
I'm not sure about locking Enthrall (the only fun mechanic of Vampire and probably the one thing it should be based around) behind a fucking 10 minutes cooldown
Vampire abilities cost blood (supposedly) to encourage him from being active

Also this is bullshit because vampire abilities are verbs without feedback message when they are up again

**Comment by Irradiation** [comment]
One thing to keep in mind is that Vampstended isn't only caused by the Vampire doing jack of shit all round. It's also caused by the (multiple) vampire(s) dying fairly quickly from various reasons, mostly being valided promptly after either being caught in the act or called out because medbay received a bloodless corpse with massive genetic damage.

**Comment by Exxion** [comment]
So you want to fix vampires doing nothing by replacing their requirement to do something with a requirement to wait longer before doing something?

File: code/game/gamemodes/vampire/vampire_powers.dm [hunk 1]
```diff
@@ -180,12 +180,12 @@
 		return
 	M.current.visible_message("<span class='warning'>[M.current.name]'s eyes flash briefly as he stares into [C.name]'s eyes</span>")
 	M.current.verbs -= /client/proc/vampire_hypnotise
-	spawn(1800)
+	spawn(2400)
```
Comment (ShiftyRail) [inline code/game/gamemodes/vampire/vampire_powers.dm:183]: While you're at it : change `2400` for `240 SECONDS` or better yet, `4 MINUTES`
Never too early to learn good formatting habits

**Comment by Allspice** [comment]
Vampire does not need a buff; what it needs is a ""rework"", which this is not.

Setting that aside, it'd perhaps be useful to you, if you got the numbers on how much blood a vamp gets from one suck, and how many uses of some relevant abilities (jaunt, screech, etc) that gives you.

**Review by ShiftyRail — COMMENTED** [review]
If you're going to make it that way, you should also look at the vampire HUD in code/game/gamemodes/vampire.dm, line 747 : 

https://github.com/vgstation-coders/vgstation13/blob/14c2349ba08b02ccfde90800066f0eacf9d5c23a/code/game/gamemodes/vampire/vampire.dm#L747-L754
Replace it with 
```dm

/mob/living/carbon/human/proc/handle_vampire()
	if(hud_used)
		if(!hud_used.vampire_blood_display)
			hud_used.vampire_hud()
			//hud_used.human_hud(hud_used.ui_style)
		hud_used.vampire_blood_display.maptext_width = WORLD_ICON_SIZE
		hud_used.vampire_blood_display.maptext_height = WORLD_ICON_SIZE
		hud_used.vampire_blood_display.maptext =<br> T:<font color='#FFFF00'>[mind.vampire.bloodtotal]</font></div>"
```
Basically it removes the "blood usable" count since it serves no purpose

**Comment by Arzion5** [comment]
@ShiftyRail It can be compared to a greytide implant that allows you to bind even implanted people to your will after a certain amount of blood has been acquired. The cooldown appears harsh, but it is suited for rounds that last for at least an hour. However, I should take into consideration that enthralled people can be quickly deconverted. Reducing the cooldown to 5 minutes will hopefully be enough. I will get to changing the code tomorrow.
@Irradiation That contributes to why vampires are hesitant on acting.
@Exxion I feel that the active blood element is restrictive, it was supposed to be balanced by the fact that you could drink from corpses, but I do not know how to fix that, and I will close this PR if anyone can fix it. Vampires doing nothing is caused by them being very hesitant due to the above-mentioned contributing causes. The cooldown increases are in place because the original cooldowns were made with the active blood price in mind.
@Allspice This is not intended to be a direct buff for vampires, I feel that active blood is a restrictive element.
For the numbers, I assume you can get around 100-160 blood from corpses, the usage for the lower-cost abilities is abundant, but enthralling still costs much with the regular costs and can be easily dispelled with holy water or a bible, and reviving will wipe out your entire active blood reserve, which the PR hasn't removed, thus you would have to drink from people again to regain your powers.
I am currently going to sleep and will try to answer to any further feedback and concerns tomorrow.

**Comment by Pathid** [comment]
@Arzion5 Please put one and only one :cl: section and it needs to be at the end of the description.

Vamps can in fact drink blood from the dead to raise their non-usable blood. It only doesn't work on monkeymen or bodies that were already cloned from, since cloning transfers the mind to the clone.

**Comment by Arzion5** [comment]
@Pathid I will try it tomorrow. If it is actually true, or if anyone else can confirm with proof, I will close the PR.

**Comment by ShiftyRail** [comment]
Yes it is : 
https://github.com/vgstation-coders/vgstation13/blob/14c2349ba08b02ccfde90800066f0eacf9d5c23a/code/game/gamemodes/vampire/vampire.dm#L417-L424

**Comment by Exxion** [comment]
I just checked to make sure and ghosting does *not* remove the mind from the corpse, so as long as the corpse hasn't had something done to it that *does* transfer the mind away, you can get blood from it. As said above, only *total* blood, not usable blood, and half as much from a living person, though. (Not that that part really matters, since I'm pretty sure you can just bite them multiple times.)
But why don't you get usable blood from corpses?

**Comment by Pathid** [comment]
@Exxion The half does matter since it's in a ``while(do_mob(src, H, 50))`` and it does this each cycle:
https://github.com/vgstation-coders/vgstation13/blob/3c28f476597abf7570f5ada0bb6cd836647cd2ae/code/game/gamemodes/vampire/vampire.dm#L428

**Comment by Kurfursten** [comment]
I played frequently during the free vamp spell days and it wasn't fun for anyone. This takes it a step further because shadowstep is now bat form so there's a lot of incentive to just idle in maint building up a bat army with your two summon spells.

**Comment by SonixApache** [comment]
Pathid is a nerd, the :cl: emoji needs to be at the *start* of the changelog

**Comment by Hoshi-chan** [comment]
Vampire has always seemed like it wants to be somewhere in between a stealth antag like changeling and a loud and visible antag like wizard, but in application, it's almost exactly like changeling: a round where fucking nothing happens until they use the screech on the shuttle, or they get ashed within the first 15 minutes of the round.

Making all abilities free is dumb; it didn't work before, and it won't work now. Vampire needs something close to a total rework. Ideally, its abilities would be a combination of free with a cooldown, blood cost with a cooldown, and blood cost with no cooldown, and there would be more of a sense of customized progression and ability choices, rather than all vampires having the same set of abilities, so a player could decide whether they wanted to build to go loud, or keep things stealthy and subtle. You know, like every other antag.

I've got more to say on the subject, but this isn't exactly the PR to pitch ideas.

**Comment by Kurfursten** [comment]
I'm not even strictly opposed to Vampires having a natural "straight" progression to make them different from other antags, but the way the vamp progression works is nonsensical. Vamps who have murderboned are effectively immune to chapel/holy water but vamps who have killed almost no one will light up shortly.

Shouldn't it be the other way around, to prevent roundstart metatests and give sec a logical way to fight back, trying to bless tiles and create protection zones with the chaplain as a key contributor like in cult?

**Comment by Reviire** [comment]
The only thing i'd really agree with from this PR, is the inclusion of abilities without blood costs. Although, blood for stronger abilities, or enhancing it would be nice.

A rework is definitely needed, shit mode.

**Comment by Arzion5** [comment]
As I have said, I will close the PR. However, I have noticed another oversight, mainly that cloning or placing the brain into an MMI will transfer the mind, making the corpse undrinkable. Vampire is currently an underwhelming gamemode, the following may be done to contribute to making them better, at least one should suffice:
- Complete ability rework to complement a bloodless mechanic
- Conversion from a fully dedicated round antagonist into a random mid-round antagonist that anyone may become at random except for the currently restricted roles. (May require antag datums.)
- Possible expansion on what can be called "Blood Magic", with shameless inspiration from the Tremere.

**Comment by SonixApache** [comment]
suggestion: help finish antag datums first, *then* overhaul antags

**Pull request closed without merging**